relay-general/src/pii/config.rs

```diff
@@ -10,6 +10,14 @@ use crate::pii::builtin::BUILTIN_RULES_MAP;
 use crate::pii::Redaction;
 use crate::processor::SelectorSpec;
 
+pub fn regex_from_user_input(pattern: &str) -> RegexBuilder {
+    // user-defined regexes should not create massive cache state.
+    let mut builder = RegexBuilder::new(pattern);
+    builder.dfa_size_limit(2048);
+    builder.size_limit(262_144);
```

Comment:
Might help, but how did you arrive at these values?

Reply:
`size_limit` we applied inconsistently before, and `dfa_size_limit` can be calculated:

* 12 threads in prod per instance
* 30k projects in cache
* 3gb per machine

```
3000 / 30000 / 12 = 0.008333 mb = 9kb
```

The default of `dfa_size_limit` is 4 MB, which is both okay for builtin rules and completely unacceptable for user-defined rules. 4 MB per thread per project would mean `4 * 30000 * 12 = 1440000 mb` max memory consumption.

Reply:
We'll likely need to get even lower as users define their own regexes, but this should be fine for now.